drivers/mysql/mysql_test.go

```diff
@@ -0,0 +1,469 @@
+
+	var result int
+	err = suite.db.QueryRow(fmt.Sprintf("SELECT COUNT(*) FROM performance_schema.metadata_locks WHERE OBJECT_TYPE = 'USER LEVEL LOCK' AND LOCK_STATUS = 'GRANTED' AND OBJECT_NAME = '%s'", advisoryLockID)).Scan(&result)
+	suite.Require().NoError(err, "should not error querying performance_schema.metadata_locks")
+	suite.Require().Equal(1, result, "advisory lock should be acquired")
+}
+
+func (suite *MysqlTestSuite) TestUnlock() {
+	driver, err := drivers.GetDriver(driverName)
+	suite.Require().NoError(err, "fetching already registered driver should not fail")
+
+	connectedDriver, err := driver.Open(testConnURL)
+	suite.Assert().NoError(err, "should not error when connecting to database from url")
+	defer func() {
+		err = driver.Close()
+		suite.Require().NoError(err, "should not error when closing the database connection")
+	}()
+
+	err = connectedDriver.Lock()
+	suite.Require().NoError(err, "should not error when attempting to acquire an advisory lock")
+
+	advisoryLockID, err := drivers.GenerateAdvisoryLockID("morph_test", defaultConfig.MigrationsTable)
+	suite.Require().NoError(err, "should not error when generating generate advisory lock id")
+
+	err = connectedDriver.Unlock()
+	suite.Require().NoError(err, "should not error when attempting to release an advisory lock")
+
+	var result int
+	err = suite.db.QueryRow(fmt.Sprintf("SELECT COUNT(*) FROM performance_schema.metadata_locks WHERE OBJECT_TYPE = 'USER LEVEL LOCK' AND LOCK_STATUS = 'GRANTED' AND OBJECT_NAME = '%s'", advisoryLockID)).Scan(&result)
+	suite.Require().NoError(err, "should not error querying performance_schema.metadata_locks")
+	suite.Require().Equal(0, result, "advisory lock should be released")
+}
+
+func (suite *MysqlTestSuite) TestAppliedMigrations() {
+	driver, err := drivers.GetDriver(driverName)
+	suite.Require().NoError(err, "fetching already registered driver should not fail")
+
+	connectedDriver, err := driver.Open(testConnURL)
+	suite.Assert().NoError(err, "should not error when connecting to database from url")
+	defer func() {
+		err = driver.Close()
+		suite.Require().NoError(err, "should not error when closing the database connection")
+	}()
+
+	err = connectedDriver.CreateSchemaTableIfNotExists()
+	suite.Require().NoError(err, "should not error when creating migrations table")
+
+	insertMigrationsQuery := fmt.Sprintf(`
+		INSERT INTO %s(Version, Name)
+		VALUES
+		       (1, 'test_1'),
+			   (3, 'test_3'),
+			   (2, 'test_2');
+	`, defaultConfig.MigrationsTable)
+	_, err = suite.testDB.Exec(insertMigrationsQuery)
+	suite.Require().NoError(err, "should not error when inserting seed migrations")
+
+	appliedMigrations, err := connectedDriver.AppliedMigrations()
+	suite.Require().NoError(err, "should not error when fetching applied migrations")
+	suite.Assert().Len(appliedMigrations, 3)
+}
+
+func (suite *MysqlTestSuite) TestApply() {
+	testData := []map[string]interface{}{
```

Comment:
Using a custom slice of `struct` here would ease the access quite a bit and avoid the casting. Any reason to use a `map[string]string` instead?

Reply:
No reason at all, amending.